chain/consensus/hierarchical/actors/subnet/subnet_actor.go

```diff
@@ -461,6 +448,11 @@ func (st *SubnetState) addStake(rt runtime.Runtime, sourceAddr address.Address,
 		// There can only be a single miner in delegated consensus.
 		if st.Consensus != hierarchical.Delegated || len(st.Miners) < 1 {
 			st.Miners = append(st.Miners, sourceAddr)
+			st.ValidatorSet = append(st.ValidatorSet, hierarchical.Validator{
+				Subnet:  address.NewSubnetID(st.ParentID, sourceAddr),
+				Addr:    sourceAddr,
+				NetAddr: params.ValidatorNetAddr,
```

Comment:
What if JoinParams is nil?

Reply:
Here `params` comes from the argument of the `Join` method. I suppose the VM takes care of passing actor correct method arguments. If there is some other piece of actor's code that incorrectly passes `params == nil` then this is a programmer's error, and `addStake` will rightfully panic.